### PR TITLE
[CI:DOCS] Documented that uidmap and guidmap are based on subgid subuid mapping

### LINKED ISSUE #6123: Document that in rootless mode, uidmap is in terms of the default mapping, not host uids
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I have a container that accesses a device as a non-root user (in the container) that I'm trying to run as a non-root user (outside of the container). To have access to the device, I need that user in the container, not root, to be mapped to my user ID, but I still need the containers entry point to be started as uid 0 (in the container).

It seemed this should be doable with --uidmap, which is documented as mapping container uids to host uids, so I kept trying variants of `--uidmap=0:1065536:1000 --uidmap=1000:1000:1 --uidmap=1001:1066536:64536` to no avail. (My user's subuid entry is 1065536:65536).)

After, a lot of poking, prodding, reading random bugs, and playing with `--userns=keep-id` (which I can't use in this case, since I still need the entry point started as container uid 0, and which wouldn't work in the general case anyway, since my host uid and the relevant container uid aren't guaranteed to match), I eventually discovered that, when running rootless, the provided uidmap is actually applied *after* the default uidmap has already been applied, so what I actually needed to specify was `--uidmap=0:1:1000 --uidmap=1000:0:1 --uidmap=1001:1001:64536` to get the mapping I wanted, which is not at all what I would expect from reading the documentation.

If this is, in fact, the intended behavior, the podman-run man page should be updated to explain how uidmap works in rootless mode. If this is not the intended behavior, I guess consider this a bug report for uidmap rather than a documentation bug. :slightly_smiling_face: 

**Additional information you deem important (e.g. issue happens only occasionally):**
If it's relevant, I'm using cgroups v2.

**Output of `podman version`:**

```
Version:            1.9.0
RemoteAPI Version:  1
Go Version:         go1.13.10
Built:              Tue May  5 17:23:34 2020
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  gitCommit: ""
  goVersion: go1.13.10
  podmanVersion: 1.9.0
host:
  arch: amd64
  buildahVersion: 1.14.8
  cgroupVersion: v2
  conmon:
    package: Unknown
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.15, commit: 1bddbf7051a973f4a4fecf06faa0c48e82f1e9e1'
  cpus: 8
  distribution:
    distribution: gentoo
    version: unknown
  eventLogger: journald
  hostname: rkjnsn-laptop-linux
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 1065536
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 1065536
      size: 65536
  kernel: 5.4.28-gentoo
  memFree: 4033044480
  memTotal: 16663584768
  ociRuntime:
    name: crun
    package: Unknown
    path: /usr/bin/crun
    version: |-
      crun version 0.10.6
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: Unknown
    version: |-
      slirp4netns version 1.0.1
      commit: 6a7b16babc95b6a3056b33fb45b74a6f62262dd4
      libslirp: 4.3.0
  swapFree: 17042067456
  swapTotal: 17179693056
  uptime: 113h 30m 26.53s (Approximately 4.71 days)
registries:
  search:
  - docker.io
  - quay.io
  - registry.fedoraproject.org
store:
  configFile: /home/rkjnsn/.config/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 0
    stopped: 2
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: Unknown
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 1.0.0
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  graphRoot: /home/rkjnsn/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 2
  runRoot: /run/user/1000/containers
  volumePath: /home/rkjnsn/.local/share/containers/storage/volumes
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Physical Gentoo machine

### PR DESCRIPTION
**Pull request by topas-rec**
* Closes #6123

Please review. This mapping is not easy to explain.

### CODE REVIEW

File: docs/source/markdown/podman-run.1.md [hunk 1]
```diff
@@ -394,6 +394,13 @@ Run the container in a new user namespace using the supplied mapping. This optio
 This option can be passed several times to map different ranges. If calling **podman run** as an unprivileged user, the user needs to have the right to use the mapping. See **subuid**(5).
 The example maps gids **0-1999** in the container to the gids **30000-31999** on the host: **--gidmap=0:30000:2000**.
 
+**Important note:** The new user namespace mapping based on **--gidmap** is based on the initial mapping made in  _/etc/subgid_  file.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:397]:
```suggestion
**Important note:** The new user namespace mapping based on **--gidmap** is based on the initial mapping made in the  _/etc/subgid_  file.   
```

File: docs/source/markdown/podman-run.1.md [hunk 2]
```diff
@@ -394,6 +394,13 @@ Run the container in a new user namespace using the supplied mapping. This optio
 This option can be passed several times to map different ranges. If calling **podman run** as an unprivileged user, the user needs to have the right to use the mapping. See **subuid**(5).
 The example maps gids **0-1999** in the container to the gids **30000-31999** on the host: **--gidmap=0:30000:2000**.
 
+**Important note:** The new user namespace mapping based on **--gidmap** is based on the initial mapping made in  _/etc/subgid_  file.   
+Assuming there is a  _/etc/subgid_  mapping **groupname:100000:65536**, then **groupname** is initially mapped to a namespace starting with
+gid **100000** for **65536** ids. From here the **--gidmap** mapping to the new namespace starts from **0** again, but is based on the initial mapping.
+Meaning **groupname** is initially mapped to gid **100000** which is referenced as **0** in the following **--gidmap** mapping. In terms of the example
+above together with the mentioned  _/etc/subgid_  mapping: The group **groupname** is mapped to group **100000** of the initial namespace then the
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:401]:
```suggestion
above: The group **groupname** is mapped to group **100000** of the initial namespace then the
```

File: docs/source/markdown/podman-run.1.md [hunk 3]
```diff
@@ -1030,6 +1037,15 @@ as an unprivileged user, the user needs to have the right to use the mapping. Se
 
 The following example maps uids 0-1999 in the container to the uids 30000-31999 on the host: **--uidmap=0:30000:2000**.
 
+**Important note:** The new user namespace mapping based on **--uidmap** is based on the initial mapping made in  _/etc/subuid_  file.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:1040]: similar changes as above

**Comment by topas-rec** [comment]
@TomSweeneyRedHat your suggestions are applied now. Thanks for reviewing.

**Comment by zhangguanzhang** [comment]
Thank you for your code contribution, you nedd sign your info in your git commit messages.

**Comment by topas-rec** [comment]
> Thank you for your code contribution, you nedd sign your info in your git commit messages.

@zhangguanzhang is this done now?

**Comment by TomSweeneyRedHat** [comment]
@topas-rec I'm not sure you signed the commit yet.  Did you do `git commit --amend-s` and then fetch, rebase and push?

**Comment by rhatdan** [comment]
@topas-rec Could you rebase this PR.  The EXT Services tests needs a rebase to pass.

**Comment by topas-rec** [comment]
I use the git integration in eclipse. Signing always failed cause the secret key was not found to the public key.
I used the terminal instead and it did not fail. I was unable to enter the commit message in multiple lines.

Is it signed now?

**Comment by topas-rec** [comment]
> @topas-rec Could you rebase this PR. The EXT Services tests needs a rebase to pass.

Should be done

**Comment by rhatdan** [comment]
@topas-rec Still has two PRs Can you merge and make sure they are signed.

**Comment by topas-rec** [comment]
I need help with signing.

I do the following and think it it correct but I cannot sign a commit:
```
tobias@tobias-pc:~/DATA/Computers/Software_development/git/podman$ gpg --list-keys
/home/tobias/.gnupg/pubring.kbx
-------------------------------
pub   rsa3072 2020-12-16 [SC] [expires: 2022-12-16]
      76831E5104F45B3949F6BB8874C4DD1712EA7974
uid           [ultimate] Tobias Boesch <tobias.boesch@googlemail.com>
sub   rsa3072 2020-12-16 [E] [expires: 2022-12-16]

tobias@tobias-pc:~/DATA/Computers/Software_development/git/podman$ git commit -a -S -m 'Documented that uidmap and guidmap are based on subgid subuid mapping * Closes #6123 Signed-off-by: Tobias Boesch <tobias.boesch@googlemail.com>'
On branch document_uid_gid_map_is_relative_to_subuid_subgid_mapping_fixes_6123
Your branch is up to date with 'origin/document_uid_gid_map_is_relative_to_subuid_subgid_mapping_fixes_6123'.

Untracked files:
	.project

nothing added to commit but untracked files present
tobias@tobias-pc:~/DATA/Computers/Software_development/git/podman$ git log --show-signature -2
commit f8be73cef24abf2b75f3185c7c8399c7721e1ce3 (HEAD -> document_uid_gid_map_is_relative_to_subuid_subgid_mapping_fixes_6123, origin/document_uid_gid_map_is_relative_to_subuid_subgid_mapping_fixes_6123)
Author: Tobias Boesch <tobias.boesch@googlemail.com>
Date:   Fri Dec 11 22:36:56 2020 +0100

    Documented that uidmap and guidmap are based on subgid subuid mapping
    
    * Closes #6123
    
    Signed-off-by: Tobias Boesch <tobias.boesch@googlemail.com>

commit a17afa931d1aa73b8657cf26de3b49841837f66d (upstream/master)
gpg: Signature made Do 17 Dez 2020 18:22:37 CET
gpg:                using RSA key 4AEE18F83AFDEB23
gpg: Can't check signature: No public key
Merge: 033336606 86335aa4a
Author: OpenShift Merge Robot <openshift-merge-robot@users.noreply.github.com>
Date:   Thu Dec 17 17:22:37 2020 +0000

    Merge pull request #8752 from baude/bindings3volumes
    
    misc bindings to podman v3
tobias@tobias-pc:~/DATA/Computers/Software_development/git/podman$ 

```

**Comment by topas-rec** [comment]
I had a problem with the created gpg key. After recreating and using `--amend` instead of `-a` at commit command (and entering the correct passphrase) it worked.

git says signed now.

```
tobias@tobias-pc:~/DATA/Computers/Software_development/git/podman$ git log --show-signature -2
commit c1d5c350984e96bbd3020f3e0c7d5ad22f3a019a (HEAD -> document_uid_gid_map_is_relative_to_subuid_subgid_mapping_fixes_6123)
gpg: Signature made Do 17 Dez 2020 22:25:54 CET
gpg:                using RSA key A1FB7D9E543E834FE0996DD02D8179D609148CFA
gpg: Good signature from "Tobias Boesch <tobias.boesch@googlemail.com>" [ultimate]
Author: Tobias Boesch <tobias.boesch@googlemail.com>
Date:   Fri Dec 11 22:36:56 2020 +0100

    Documented that uidmap and guidmap are based on subgid subuid mapping * Closes #6123 Signed-off-by: Tobias Boesch <tobias.boesch@googlemail.com>

commit a17afa931d1aa73b8657cf26de3b49841837f66d (upstream/master)
gpg: Signature made Do 17 Dez 2020 18:22:37 CET
gpg:                using RSA key 4AEE18F83AFDEB23
gpg: Can't check signature: No public key
Merge: 033336606 86335aa4a
Author: OpenShift Merge Robot <openshift-merge-robot@users.noreply.github.com>
Date:   Thu Dec 17 17:22:37 2020 +0000

    Merge pull request #8752 from baude/bindings3volumes
    
    misc bindings to podman v3
tobias@tobias-pc:~/DATA/Computers/Software_development/git/podman$ 

```

**Comment by topas-rec** [comment]
Additionally I had to change the email from gmail to googlemail in github.

**Comment by rhatdan** [comment]
The validate can be a little picky, it wants a blank line between your signature and the tittle of your PR.  The title needs to be less then 80 chars also.

The PR can also not have any whitespace.

**Comment by topas-rec** [comment]
> The validate can be a little picky, it wants a blank line between your signature and the tittle of your PR. The title needs to be less then 80 chars also.
> 
> The PR can also not have any whitespace.

Okay, is this something that prevents this PR from beeing merged?

**Comment by rhatdan** [comment]
Yes all PRs have to pass the tests to get merged.

  - FAIL - has whitespace errors. See `git show --check 2ca3bc213d3a2a9e16260abe2350551ad7ac85ce`.
  - FAIL - does not have a valid DCO
  - FAIL - commit subject exceeds 90 characters

**Comment by topas-rec** [comment]
> FAIL - has whitespace errors. See git show --check \<commit id\>

image

I wanted to add a new line like explained [here](https://stackoverflow.com/questions/33191744/how-to-add-new-line-in-markdown-presentation).
This is not an issue I think. Now the tests are passed, but I intended to have a new line in the documentation - which is now gone

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -3,10 +3,10 @@
 
 We'd love to have you join the community! Below summarizes the processes
 that we follow.
-
+asd
 ## Topics
 
-* [Reporting Issues](#reporting-issues)
+* [Reporting Issueqweqwes](#reporting-issues)
```
Comment (rhatdan) [inline CONTRIBUTING.md:9]: Looks like typos have snuck in.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -3,10 +3,10 @@
 
 We'd love to have you join the community! Below summarizes the processes
 that we follow.
-
+asd
```
Comment (rhatdan) [inline CONTRIBUTING.md:6]: Typo?

**Review by rhatdan — APPROVED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (topas-rec) [reply CONTRIBUTING.md]: > Typo?

Yes. Rermoved

File: CONTRIBUTING.md (hunk 4 above)
Comment (topas-rec) [reply CONTRIBUTING.md]: Removed

**Comment by topas-rec** [comment]
Thanks for your help so far.

**Comment by rhatdan** [comment]
/lgtm
Thanks @topas-rec

**Pull request merged**